modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalBasicWithFieldName.tsx

```diff
@@ -0,0 +1,154 @@
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayAlert from '@clayui/alert';
+import ClayButton from '@clayui/button';
+import ClayForm from '@clayui/form';
+import ClayModal from '@clayui/modal';
+import React, {useState} from 'react';
+
+import useForm from '../hooks/useForm';
+import Input from './form/Input';
+
+interface IProps extends React.HTMLAttributes<HTMLElement> {
+	apiURL: string;
+	inputId: string;
+	label: string;
+	observer: any;
+	onClose: () => void;
+}
+
+type TInitialValues = {
+	name: {
+		[key: string]: string;
+	};
+};
+
+const defaultLanguageId = Liferay.ThemeDisplay.getDefaultLanguageId();
+
+const labels = {
+	newLayout: Liferay.Language.get('new-layout'),
+	newView: Liferay.Language.get('new-view'),
+};
+
+const ModalBasicWithFieldName: React.FC<IProps> = ({
+	apiURL,
+	inputId,
+	label,
+	observer,
+	onClose,
+}) => {
+	const initialValues: TInitialValues = {
+		name: {
+			[defaultLanguageId]: '',
+		},
+	};
+	const [error, setError] = useState<string>('');
+
+	const onSubmit = async ({name}: TInitialValues) => {
+		const response = await Liferay.Util.fetch(apiURL, {
+			body: JSON.stringify({
+				name: {
+					[defaultLanguageId]: name,
+				},
+			}),
+			headers: new Headers({
+				'Accept': 'application/json',
+				'Content-Type': 'application/json',
+			}),
+			method: 'POST',
+		});
+
+		if (response.status === 401) {
+			window.location.reload();
+		}
+		else if (response.ok) {
+			onClose();
+
+			window.location.reload();
+		}
+		else {
+			const {
+				title = Liferay.Language.get('an-error-occurred'),
+			} = await response.json();
+
+			setError(title);
+		}
+	};
+
+	const validate = ({name}: TInitialValues) => {
+		const errors: any = {};
+
+		if (!name[defaultLanguageId] && typeof name !== 'string') {
+			errors.name = Liferay.Language.get('required');
+		}
+
+		return errors;
+	};
+
+	const {errors, handleChange, handleSubmit, values} = useForm({
+		initialValues,
+		onSubmit,
+		validate,
+	});
+
+	return (
+		<>
+			<ClayModal observer={observer}>
+				<ClayForm onSubmit={handleSubmit}>
+					<ClayModal.Header>
+						{label === 'New Layout'
+							? labels.newLayout
+							: labels.newView}
+					</ClayModal.Header>
+
+					<ClayModal.Body>
+						{error && (
+							<ClayAlert displayType="danger">{error}</ClayAlert>
+						)}
+
+						<Input
+							error={errors.name}
+							id={inputId}
+							label={Liferay.Language.get('name')}
+							name="name"
+							onChange={handleChange}
+							required
+							value={values.name[defaultLanguageId]}
+						/>
+					</ClayModal.Body>
+
+					<ClayModal.Footer
+						last={
+							<ClayButton.Group key={1} spaced>
+								<ClayButton
+									displayType="secondary"
+									onClick={() => onClose()}
```

Comment:
It seems you can directly call the `onClose`:
```suggestion
									onClick={onClose}
```